clustering/infinispan/extension/src/main/java/org/jboss/as/clustering/infinispan/subsystem/remote/RemoteConfigurationBuilder.java

```diff
@@ -0,0 +1,177 @@
+    }
+
+    @Override
+    public ServiceName getServiceName() {
+        return CONFIGURATION.getServiceName(this.address);
+    }
+
+    @Override
+    public Builder<Configuration> configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        this.connectionTimeout = Attribute.CONNECTION_TIMEOUT.getDefinition().resolveModelAttribute(context, model).asInt();
+        this.defaultRemoteCluster = Attribute.DEFAULT_REMOTE_CLUSTER.getDefinition().resolveModelAttribute(context, model).asString();
+        this.keySizeEstimate = Attribute.KEY_SIZE_ESTIMATE.getDefinition().resolveModelAttribute(context, model).asInt();
+        this.maxRetries = Attribute.MAX_RETRIES.getDefinition().resolveModelAttribute(context, model).asInt();
+        this.protocolVersion = Attribute.PROTOCOL_VERSION.getDefinition().resolveModelAttribute(context, model).asString();
+        this.socketTimeout = Attribute.SOCKET_TIMEOUT.getDefinition().resolveModelAttribute(context, model).asInt();
+        this.tcpNoDelay = Attribute.TCP_NO_DELAY.getDefinition().resolveModelAttribute(context, model).asBoolean();
+        this.tcpKeepAlive = Attribute.TCP_KEEP_ALIVE.getDefinition().resolveModelAttribute(context, model).asBoolean();
+        this.valueSizeEstimate = Attribute.VALUE_SIZE_ESTIMATE.getDefinition().resolveModelAttribute(context, model).asInt();
+
+        this.clusters.clear();
+        if (model.hasDefined(RemoteClusterResourceDefinition.WILDCARD_PATH.getKey())) {
+            for (Property property : model.get(RemoteClusterResourceDefinition.WILDCARD_PATH.getKey()).asPropertyList()) {
+                String clusterName = property.getName();
+                List<InjectedValueDependency<OutboundSocketBinding>> valueDependencies = StringListAttributeDefinition.unwrapValue(context, RemoteClusterResourceDefinition.Attribute.SOCKET_BINDINGS.resolveModelAttribute(context, property.getValue())).stream()
+                        .map(binding -> new InjectedValueDependency<>(CommonUnaryRequirement.OUTBOUND_SOCKET_BINDING.getServiceName(context, binding), OutboundSocketBinding.class))
+                        .collect(Collectors.toList());
+                this.clusters.put(clusterName, valueDependencies);
+            }
+        }
+
+        return this;
+    }
+
+    @Override
+    public ServiceBuilder<Configuration> build(ServiceTarget target) {
+        ServiceBuilder<Configuration> builder = target.addService(this.getServiceName(), new ValueService<>(this))
+                .addDependency(RemoteCacheContainerComponent.CONNECTION_POOL.getServiceName(this.address), ConnectionPoolConfiguration.class, this.connectionPool)
+                .addDependency(RemoteCacheContainerComponent.NEAR_CACHE.getServiceName(address), NearCacheConfiguration.class, this.nearCache)
+                .setInitialMode(ServiceController.Mode.PASSIVE)
```

Comment:
This should be ON_DEMAND.

Reply:
Done. Original when the PR was open we were still installing services as passive.